bin/arch/debian-sw-probe/debian-sw-probe-ATLAS.sh

```diff
@@ -93,15 +93,18 @@ do
 done
 
 # Make sure /var/atlasdata is mounted on tmpfs. This is needed for 
-# devices that run for small/cheap flash. We may need a switch to turn
+# devices that run for small/cheap flash. Set CHECK_TMPFS to no to turn
 # this off
-while :
-do
-	if mount | grep -q '/var/atlasdata.*tmpfs'
-	then
-		# ready to go
-		break
-	fi
-	echo '/var/atlasdata is not mounted' >&2
-	sleep 60
-done
+CHECK_TMPFS=${CHECK_TMPFS:-yes}
+if [ "$CHECK_TMPFS" = "yes" ]; then
+	while :
```

Comment:
Nit: when looking at this, I might consider avoiding the extra nesting by something like
```bash
while [ "${CHECK_TMPFS}" != no ]
do
    #unchanged inside
done
```
but perhaps it would be ugly, kind of.

Reply:
Yep, I think using a standalone `if` kind of expresses the intention better here.

Reply:
I'm a bit confused how this works. Where is this variable set? There is a simple config file (config.sh), but this patch doesn't seem to pick up the variable from that

Reply:
@PhilipHomburg I'm a little confused about the `config.sh`. It seems to be only containing permanent, platform specific configs that ship with the package, and I thought whether we wait for the tmpfs is more or less a user-configurable thing and if they don't need this feature they can just change the settings at any time and restart the atlas service. It doesn't need to be explicitly set in a config file; it can be pull in using inheritance (thus a default value is given in my code). If you think it is apporpriate, I can add the variable to `config.sh`.

My first thought on this feature is that I maintain a docker image of the probe software based on debian, and there are certain commercial OSes that provide docker functionality though a GUI that doesn't allow mounting a tmpfs so the atlas service will just hang there, so if there is a switch to disable that wait, the probe can be run on these devices without a hassle.

Reply:
Ah, it is too warm here. I meant config.txt

Reply:
The mount of tmpfs is meant to have a  safe default on a raspberry pi and similar. It is not a problem to disable it on bigger hardware

Reply:
@PhilipHomburg Now I have more questions :)

1) As in

https://github.com/RIPE-NCC/ripe-atlas-software-probe/blob/58f21c8f1cdc465fa7e8ebb0678529b2519dce29/bin/ATLAS#L19-L33

, config.txt is read after `debian-sw-probe-ATLAS.sh`. I suggest we move L19 after L33 but this will require `common.sh` to not depend on `debian-sw-probe-ATLAS.sh` (which it is now but I'd like to ask if it is intended or there is possibility this change will break some assumptions).

2) The current code for parsing `config.txt`

https://github.com/RIPE-NCC/ripe-atlas-software-probe/blob/58f21c8f1cdc465fa7e8ebb0678529b2519dce29/bin/ATLAS#L26-L33

is not scalable at all. (i.e. if we ever want to add another option, we are going to start repeating ourselves and make the script look bad.) Is it possible that we make `config.txt` a "posix-compatible" `key=value` format so we can simply source it? Or maybe use some better parsing method?

Reply:
My current idea is that there will be just a few variables in config.txt. So we can make a simple function that gets the name of a variable as argument and maybe a default value and then returns the result. I want to avoid sourcing things as much as possible. 

I can refactor the current code when I have time

Reply:
Now in master, the config variable CHECK_ATLASDATA_TMPFS can be set to 'no' to disable the tmpfs check.